src/main/kotlin/nl/avisi/structurizr/site/generatr/site/views/Markdown.kt

```diff
@@ -34,18 +38,20 @@ private fun markdownToHtml(pageViewModel: PageViewModel, markdown: MarkdownViewM
     val renderer = HtmlRenderer.builder(options)
         .linkResolverFactory(CustomLinkResolver.Factory(pageViewModel))
         .build()
-    val document = parser.parse(markdown.markdown)
+    val markDownDocument = parser.parse(markdown.markdown)
+    val html = renderer.render(markDownDocument)
 
-    return renderer.render(document)
+    return Jsoup.parse(html)
+        .apply { body().transformEmbeddedDiagramElements(pageViewModel) }
```

Comment:
The post processing here is not really cool, though I haven't found a way how to this with a flexmark plugin.

Reply:
Fine by me. It's an isolated thing, that doesn't impact consumers of this class.